api/v3/Engage/Signpetition.php

```diff
@@ -109,6 +109,9 @@ function _civicrm_api3_engage_signpetition_process($params) {
       CRM_Gpapi_Processor::addToGroup($contact_id, 'Community NL');
     }
 
+    //remove "Opt Out" and "do not email"
+    CRM_Gpapi_Processor::enableSubscription($contact_id);
```

Comment:
This should happen only in the `if` above.

Reply:
Above you can see comment (GP-463: "der Group "Donation Info" Eintrag soll immer gesetzt werden..."). If contact is always added to group, that's why we always call 'enableSubscription' method.

Reply:
If you want we can move `enableSubscription` to `if` body.

Reply:
As discussed in Redmine: Yes, please move it to the `newsletter` if block.